.github/workflows/tiledb-go.yml

```diff
@@ -121,12 +121,91 @@ jobs:
     - name: Test TileDB-Go
       run: go test -v ./...
 
+  Linux_Experimental_Test:
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        # Will be checking following versions
+        go: ["1.16", "1.17"]
+    steps:
+
+    # Checks out repository
+    - uses: actions/checkout@v2
+
+    # Downloads TileDB-Core from release assets and install
+    - name: Run TileDB install script
+      run: ./.github/scripts/install_tiledb_source_linux.sh
+      shell: bash
+
+    # Following action sets up Go and uses the strategy matrix to test on
+    # specific versions
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ matrix.go }}
+
+    - name: Cache dependencies
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.cache/go-build
+          ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
```

Comment:
You should likely add -experimental at the end of this so that we don’t contaminate our non-experimental caches with our experimental caches.